app/pages/lab-pages-editor/DataManager.jsx

```diff
@@ -0,0 +1,97 @@
+/*
+Data Manager
+Responsible for fetching, updating, and saving Workflow resources from/to
+the Panoptes service.
+ */
+
+import React, { useEffect, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import apiClient from 'panoptes-client/lib/api-client';
+import { WorkflowContext } from './context.jsx';
+
+function DataManager({
+  // key: to ensure DataManager renders FRESH (with states reset) whenever workflowId changes, use <DataManager key={workflowId} ... />
+  children,
+  workflowId
+}) {
+  let initialised = false;
+  const [apiData, setApiData] = useState({
+    workflow: null,
+    status: 'ready'
+  });
+
+  async function fetchWorkflow() {
+    try {
+      setApiData({
+        workflow: null,
+        status: 'fetching'
+      });
+
+      const wf = await apiClient.type('workflows').get(workflowId);
+      if (!wf) throw new Error('No workflow');
+
+      setApiData({
+        workflow: wf,
+        status: 'ready'
+      });
+    } catch (err) {
+      console.error('DataManager: ', err);
+      setApiData({
+        workflow: null,
+        status: 'error'
+      });
+    }
+  }
+
+  // Fetch workflow when the component loads for the first time.
+  // See notes about 'key' prop, to ensure states are reset:
+  // https://react.dev/learn/you-might-not-need-an-effect#resetting-all-state-when-a-prop-changes
+  // Also see general pattern notes:
+  // https://react.dev/reference/react/useEffect#fetching-data-with-effects
+  useEffect(() => {
+    if (!initialised) fetchWorkflow();
+    initialised = true;
+    return () => { initialised = false; };
+  }, [workflowId]);
```

Comment:
```suggestion
  }, [fetchWorkflow, initialised]);
```

The hook's external dependencies are now `initialised` and `fetchWorkflow`. `fetchWorkflow` will be a new function on every render, so be careful about that, as it will lead to your effect hook possibly triggering on each render.

The React docs recommend declaring your data-fetching function inside the effect hook, so that a new function is only created each time that the effect runs.

Reply:
I purposely removed `initialised` in case it caused the useEffect hook to go into an infinite update loop. I'll try re-adding and test it out.

Reply:
I've removed `initialised` altogether.